addons/website_slides/controllers/main.py

```diff
@@ -524,6 +648,32 @@ def slide_category_search_read(self, fields, domain):
             'can_create': can_create,
         }
 
+    @http.route('/slides/add_category', type="json", website=True, auth="user")
+    def add_category(self, channel_id, name, **kw):
+        channel = request.env['slide.channel'].browse(channel_id)
+        request.env['slide.category'].create({
+            'name': name,
+            'channel_id': channel.id
+        })
+
+        return {'url': "/slides/%s" %(slug(channel))}
+
+    #Not using the /web/dataset/resequence route as a slide can be dragged into another category at the same time
+    @http.route('/slides/resequence_slides', type="json", website=True, auth="user")
+    def resequence_slides(self, slides_data=None, **kw):
+        channel_user = None
+        for slide in slides_data:
+            s = request.env['slide.slide'].browse(slide['id'])
+            if not channel_user:
+                channel_user = s.channel_id.user_id
+                if request.env.user != channel_user:
+                    return {'error': 'Only the responsible of the channel can edit it'}
+            s.write({
```

Comment:
And please don't use single letter variables (except maybe for lambdas), it's a little confusing.